app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts

```diff
@@ -62,6 +79,10 @@ export class WebhookStage {
     this.stage.statusUrlResolution = this.viewState.statusUrlResolution;
   }
 
+  public customHeaderCount(): number {
+    return this.stage.customHeaders ? Object.keys(this.stage.customHeaders).length : 0;
```

Comment:
Can you verify this still returns `0` when the last custom header is removed? I'm wondering if angular puts a `$$hashkey` on the `customHeaders` object when it iterates over it in the UI above, and if that gets picked up by the `Object.keys` call.